### ISSUE TITLE
Fuzzing results in index out of bounds

### ISSUE THREAD

**Issue by manunio** [fancy-regex/fancy-regex#122]
#### Description
Hi, Running fuzz_parser results in following error.
`thread '<unnamed>' panicked at 'index out of bounds: the len is 5 but the index is 6', /home/maxx/dev/security/oss-fuzz-projects/fancy-regex/src/parse.rs:774:19`

#### Steps to reproduce
```rust
#[test]
fn fuzz_3() {
    p(r"(?()^");
}
```

```sh
❯ cargo test parse::tests::fuzz_3
```

```
Compiling fancy-regex v0.12.0 (/home/maxx/dev/security/oss-fuzz-projects/fancy-regex)
Finished test [unoptimized + debuginfo] target(s) in 2.78s
Running unittests src/lib.rs (target/debug/deps/fancy_regex-fcae9c14f84f520b)

running 1 test
test parse::tests::fuzz_3 ... FAILED

failures:

---- parse::tests::fuzz_3 stdout ----
thread 'parse::tests::fuzz_3' panicked at 'index out of bounds: the len is 5 but the index is 6', src/parse.rs:774:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    parse::tests::fuzz_3

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 73 filtered out; finished in 0.00s

error: test failed, to rerun pass `--lib`

```

**Comment by robinst** [comment]
Thanks, fix is in review:
* https://github.com/fancy-regex/fancy-regex/pull/125

**Comment by joshfried-aws** [comment]
Hi @robinst this still doesnt handle the case where there are control characters, correct?

**Comment by robinst** [comment]
@joshfried-aws It does. I tested that one, but the cause is the same as the others, the conditional `(?(`.

**Comment by robinst** [comment]
Thanks, also fixed :).

**Comment by joshfried-aws** [comment]
@robinst thank you for this. Any idea when the next release will be live so I can update my crate that has a dependency on fancy-regex?

**Comment by robinst** [comment]
> @robinst thank you for this. Any idea when the next release will be live so I can update my crate that has a dependency on fancy-regex?

Yep, I'm planning to release this in the next few days.

**Comment by robinst** [comment]
Released in 0.13.0 now: https://github.com/fancy-regex/fancy-regex/blob/0.13.0/CHANGELOG.md#0130---2023-12-22
